File: study/trees/bfs/bfs_q.cpp
#include <cstddef>
#include <iostream>
#include <queue>

class Node { 
public: 
  int key;  
  Node * left = NULL;
  Node * right = NULL;

  Node(int v) {
    key = v;
    left = NULL;
    right = NULL;
  }
};

Node * create_tree() { 
  Node * root = new Node(10); 
  root->left = new Node(11);
  root->right = new Node(12);
  root->left->left = new Node(13);
  root->left->right = new Node(14); 

  return root; 

}

void showq(std::queue<int> qg){ 
  while (!qg.empty()) { 
    std::cout << qg.front() << "\t"; 
    qg.pop(); 
  }
  std::cout << "\n";
}

int main() {
  std::queue<int> qdfs; 
  Node * tree = create_tree();

  return 0; 
}

File: study/trees/bfs/bfs_q.py
from collections import deque


class Node:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None

def bfs(root):
    if not root:
        return
    queue = [root]
    l = [[root.val]]
    
    while queue:
        s = queue.pop(0)
        la = []
        if s.left:
            queue.append(s.left)
        if s.right:
            queue.append(s.right)
        
    print(l)


def main():
    root = Node(1)
    root.left = Node(2)
    root.right = Node(3)
    root.right.left = Node(6)
    root.right.right = Node(7)
    root.left.left = Node(4)
    root.left.right = Node(5)
    bfs(root)


if __name__ == "__main__":
    main()
